cyberdrop_dl/managers/client_manager.py

```diff
@@ -120,11 +120,16 @@ async def check_http_status(
         """Checks the HTTP status code and raises an exception if it's not acceptable."""
         status = response.status
         headers = response.headers
-
-        if download and headers.get("ETag") in DOWNLOAD_ERROR_ETAGS:
+        # for downloads
+        if not download:
+            pass
+        elif headers.get("ETag") in DOWNLOAD_ERROR_ETAGS:
             message = DOWNLOAD_ERROR_ETAGS.get(headers.get("ETag"))
             raise DownloadError(HTTPStatus.NOT_FOUND, message=message, origin=origin)
+        elif HTTPStatus.OK <= status < HTTPStatus.BAD_REQUEST:
+            return
```

Comment:
for downloads data should not be read for successful requests to prevent eof errors 

```
    @classmethod
    async def check_http_status(
        cls,
        response: ClientResponse,
        download: bool = False,
        origin: ScrapeItem | URL | None = None,
    ) -> None:
        """Checks the HTTP status code and raises an exception if it's not acceptable."""
        status = response.status
        headers = response.headers

        if download and headers.get("ETag") in DOWNLOAD_ERROR_ETAGS:
            message = DOWNLOAD_ERROR_ETAGS.get(headers.get("ETag"))
            raise DownloadError(HTTPStatus.NOT_FOUND, message=message, origin=origin)

        if HTTPStatus.OK <= status < HTTPStatus.BAD_REQUEST:
            return

        if any(domain in response.url.host for domain in ("gofile", "imgur")):
            with contextlib.suppress(ContentTypeError):
                JSON_Resp: dict = await response.json()
                if "status" in JSON_Resp and "notFound" in JSON_Resp["status"]:
                    raise ScrapeError(HTTPStatus.NOT_FOUND, origin=origin)
                if "data" in JSON_Resp and "error" in JSON_Resp["data"]:
                    raise ScrapeError(JSON_Resp["status"], JSON_Resp["data"]["error"], origin=origin)

        response_text = None
        with contextlib.suppress(UnicodeDecodeError):
            response_text = await response.text()

        if response_text:
            soup = BeautifulSoup(response_text, "html.parser")
            if cls.check_ddos_guard(soup):
                raise DDOSGuardError(origin=origin)

        status = status if headers.get("Content-Type") else CustomHTTPStatus.IM_A_TEAPOT
        message = "No content-type in response header" if headers.get("Content-Type") else None

        raise DownloadError(status=status, message=message, origin=origin)
```